nemo/collections/nlp/data/common/sequence_to_sequence_dataset.py

```diff
@@ -146,76 +154,136 @@ def __init__(
             max_src_seq_length=max_src_seq_length,
             max_tgt_seq_length=max_tgt_seq_length,
         )
+        self.seed = seed
+        self.max_num_samples = max_num_samples
+        logging.info(f'Desired number of samples : {self.max_num_samples}')
+        logging.info(f'Source Dataset Length : {len(self.src_indexed_dataset)}')
+        logging.info(f'Target Dataset Length : {len(self.tgt_indexed_dataset)}')
 
     def __len__(self):
-        return len(self.src_dataset)
+        if self.max_num_samples is None:
+            return len(self.src_indexed_dataset)
+        else:
+            return self.max_num_samples
```

Comment:
What if `self.max_num_samples > len(self.src_indexed_dataset)` ?

Reply:
That's fine. `build_samples_mapping()` will then "upsample" your data to the provided `max_num_samples` amount to turn everything into single epoch training like we do for pre-training.